### PR TITLE
[nrfconnect] Automatically build OTA image

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
We need a way to build Matter OTA image as part of the build process.

#### Change overview
1. Add certain software configuration, such as vendor ID, product ID etc, to Kconfig.
2. Use values of the Kconfig variables in existing CHIP project configuration.
3. Add two more Kconfig variables: OTA_IMAGE_BUILD and OTA_IMAGE_EXTRA_ARGS. The first one enables building    OTA image file using the ota_image_tool.py script, and the second one allows to pass some extra arguments to that script.

#### Testing
Build only - verified that the generate image is valid using `ota_image_tool.py` script.

### CODE REVIEW

File: config/zephyr/Kconfig [hunk 1]
```diff
@@ -34,6 +34,43 @@ menuconfig CHIP
...
+	  accept a software update whose version is greater than the current one.
+
+config CHIP_DEVICE_SOFTWARE_VERSION_STRING
+	string "Device software version string"
+	default "prerelease"
+	help
+	  A string between 1 and 64 characters that provides a user-friendly
+	  description of the numeric software version specified in
+	  CHIP_DEVICE_SOFTWARE_VERSION. It is used in various CHIP areas, such as
+	  the Basic Information cluster or OTA (Over-the-air update) image header.
```
Comment (eve-cxrp) [inline config/zephyr/Kconfig:72]: Question: Is this not redundant information which is also part of the ChipProjectConfig? If yes, is there a way to not introduce this redundancy. This leads to errors during releases.

File: config/nrfconnect/chip-module/CMakeLists.txt [hunk 2]
```diff
@@ -28,6 +28,7 @@
 if (CONFIG_CHIP)
 
 include(ExternalProject)
+include(../../zephyr/ota-image.cmake)
```
Comment (eve-cxrp) [inline config/nrfconnect/chip-module/CMakeLists.txt:31]: Nit: don't know if ota-image is good name. maybe it should be more specific. matter-ota-image.cmake ?

File: config/zephyr/Kconfig (hunk 1 above)
Comment (Damian-Nordic) [reply config/zephyr/Kconfig:72]: Yes, if you don't specify these values in `CHIPProjectConfig.h`, they will be automatically set based on the Kconfig values. Basically, having these values available at the CMake level allows us to set both the C-defines and call the Python script to generate the OTA image.

File: config/nrfconnect/chip-module/CMakeLists.txt (hunk 2 above)
Comment (Damian-Nordic) [reply config/nrfconnect/chip-module/CMakeLists.txt:31]: Given that the module is inside the Matter core directory, can it be confused with other OTA image? If so, I can rename it, sure.

File: config/zephyr/Kconfig (hunk 1 above)
Comment (eve-cxrp) [reply config/zephyr/Kconfig:72]: Ok, thanks for the clarification 🙂

File: config/nrfconnect/chip-module/CMakeLists.txt (hunk 2 above)
Comment (eve-cxrp) [reply config/nrfconnect/chip-module/CMakeLists.txt:31]: Indeed you are right. I was just looking on the path and `zephyr/ota-image.cmake` just leads to a different association in my head. I was thinking that this `ota-image.cmake` was related to `zephyr`.

**Review by ghost — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: platform specific change

**Pull request merged**